src/status_im/common/biometric/utils.cljs

```diff
@@ -0,0 +1,36 @@
+(ns status-im.common.biometric.utils
+  (:require
+    [native-module.core :as native-module]
+    [react-native.platform :as platform]
+    [status-im.constants :as constants]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+(def android-device-blacklisted?
+  (and platform/android? (= (:brand (native-module/get-device-model-info)) "bannedbrand")))
+
+(defn get-label-by-type
+  [biometric-type]
+  (condp = biometric-type
+    constants/biometrics-type-android (i18n/label :t/biometric-fingerprint)
+    constants/biometrics-type-face-id (i18n/label :t/biometric-faceid)
+    (i18n/label :t/biometric-touchid)))
+
+(defn get-icon-by-type
+  [biometric-type]
+  (condp = biometric-type
+    constants/biometrics-type-face-id :i/face-id
+    :i/touch-id))
+
+;; NOTE: move to utils.re-frame?
+(defn handle-cb
```

Comment:
Do we really need this function? Imo it's just hiding away something easy enough to add and makes it a bit more confusing to read.
Curious to know why do we need it? 🤔

Reply:
The function was added to make it easier to pass callbacks such as `on-success`/`on-fail`/etc. both as vectors and functions, as per the [guidelines](https://github.com/status-im/status-mobile/blob/d1c8f6411d16d3ae74dd1b480493d6dcf9e00108/doc/new-guidelines.md?plain=1#L462) _(e.g. `[:do-something with-me]` and `#(rf/dispatch [:do-something with-me])`)_, by abstracting away the decision on how the callback is handled. 

The logic here is simple, but it was repeating over and over and seemed better abstracted away (not always the case, but hold the pitchforks ✋). I feel like if we want to use the approach above, would be good to make it as easy as possible, otherwise we'll just end up doing what's easier i.e. always using callbacks as functions. Another reason is having the logic in one place, for which we can have tests + schema and avoid these checks for callbacks everywhere with `(when on-fail (on-fail))` and `(if (vector? on-fail) (rf/dispatch (conj on-fail error)) ...)`.

As far as I see, would be good to use the vector approach whenever possible as it makes testing easier. I initially planned to write unit tests for standard-auth, for which this would've helped, but ended up writing integration tests (and honestly forgot about the unit ones 😆).

So I guess the questions is, do we want to pass vectors instead of functions to effects (where possible) and whether the abstraction above is justified?

Reply:
> As far as I see, would be good to use the vector approach whenever possible as it makes testing easier.

It's not that clear cut @clauxx. Vectors will force a lot of code in the repo to be refactored to dispatch to a separate `on-success` event or `on-error` event, but to what benefit if most don't have tests yet? Just being realistic. Although I'm a proponent for it since I changed our re-frame effect `:json-rpc/call` to do that, some people will find this idea inconvenient for implementing all effects in this repo. During development I find very useful to do the following, which would be impossible if we supported only the vector syntax:

```clojure
:on-success #(tap> [:success %])
:on-error #(tap> [:error %])
```

The pattern to dispatch and conj the result to the passed event vector was created way before status-mobile and it's used, for example, by the known library https://github.com/day8/re-frame-http-fx.

------

But in essence, I fully endorse the idea to have effects receiving event vectors as our first go solution. This kind of improvement should probably come as a separate PR, so that others can hear about the improvements and understand new guidelines. As @J-Son89 pointed out, I also think this kind of utility shouldn't live isolated in a biometric utility.

Reply:
makes sense it being outside the scope of this PR. I used the `:json-rpc/call` effect for reference as I remembered seeing vectors passed somewhere, so I assumed it was already decided as the way to go. Will change it to using callbacks for now.